compiler/luci/pass/src/RemoveNoEffectSlicePass.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveNoEffectSlicePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/// @brief Return value in CircleConst.
+/// @details Return value in position on CircleConst with int64 format.
+///          Begin must be larger than or equal to 0.
+///          Size must be larger than or equal to -1.
+///          Throw an error on Invalid node or invalid node type.
+int64_t value_from_circle_const(const luci::CircleConst *node, uint32_t idx)
+{
+  if (node->rank() != 1 || node->dim(0).value() <= idx)
+    throw std::runtime_error("Invalid constant node on slice");
+  if (node->dtype() == loco::DataType::S64)
+    return node->at<loco::DataType::S64>(idx);
+  else if (node->dtype() == loco::DataType::S32)
+    return static_cast<int64_t>(node->at<loco::DataType::S32>(idx));
+  throw std::runtime_error("Invalid constant type on slice");
+}
+
+bool remove_no_effect_slice(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleSlice *>(node);
+  if (target_node == nullptr)
+    return false;
+
+  auto begin_const = dynamic_cast<luci::CircleConst *>(target_node->begin());
+  if (begin_const == nullptr)
+    return false;
+
+  auto size_const = dynamic_cast<luci::CircleConst *>(target_node->size());
+  if (size_const == nullptr)
+    return false;
+
+  // Check input output shape.
+  auto input_node = loco::must_cast<luci::CircleNode *>(target_node->input());
+  for (uint32_t i = 0; i < input_node->rank(); i++)
+  {
+    if (value_from_circle_const(begin_const, i) != 0)
+      return false;
+
+    int64_t size_value = value_from_circle_const(size_const, i);
+    if (size_value == -1)
+      continue;
+
+    if (input_node->shape_signature().rank() != 0 && input_node->shape_signature().dim(i) == -1)
+      return false;
+
+    if (size_value != static_cast<int64_t>(input_node->dim(i).value()))
+      return false;
+  }
+  replace(target_node).with(input_node);
+  return true;
+}
+
+} // namespace
+
+namespace luci
+{
+/**
+ *   BEFORE
+ *      |
+ * [CircleNode]
```

Comment:
This diagram seems there is a `BEFORE` node and this is connected to  `[CircleNode]`.
It would be better to use indentation to distinghish this.

Reply:
Oh, I will fix it.